servicetalk-concurrent-api/docs/modules/ROOT/pages/blocking-implementation.adoc

```diff
@@ -20,77 +21,186 @@ Everything inside ServiceTalk is somehow connected to one of the three asynchron
+----
 
-ServiceTalk concurrency APIs defines which thread will be used by any asynchronous source for Part 2.
+<1> `toFuture().get` begins by calling `subscribe(Subscriber)`. Executing on the application thread, execution flows up
+the operator chain towards the source; `map` -> `publishOn` -> `map` -> `Publisher.range(1, 10)`.
 
-== Executor and asynchronous sources
+<2> Still executing on the application thread, `Range` will call `Subscriber.onSubscribe(Subscription)` on the
+`Subscriber`. This flows back down the operator chain, `Range` -> `map` -> `publishOn` -> `map` -> `toFuture`.
 
-ServiceTalk enforces that each asynchronous source **MUST** be associated with an `Executor` which is used to interact
-with its `Subscriber`. The below diagram illustrates the interaction between an asynchronous source, its `Subscriber`,
-its operators, and the `Executor`.
+<3> When `subscribe()` is complete, `toFuture()` will call `Subscription.request(Long.MAX_VALUE)`. This will cause
+`Range` to publish synchronously via `onNext(element)` nine items, the integers "`1`" through "`9`".
 
-image::executor-thread-selection.png[Executor inference]
+<4> Each `onNext` flows down the operator chain, `Range` -> `map` -> `publishOn` (offloads on to `publishExecutor`
+thread) -> `map` -> `toFuture` where the element is collected for the result. For each offloaded item a thread of
+`publishExecutor` will be used for executing the second `map` operator and final collect operation.
 
-The above interaction provides all operators and the source an opportunity to piggy-back on the same `Executor` for all
-interactions with the `Subscriber`. Any operator in a chain has two options while accepting a `Subscriber`:
+<5> The application thread will continue to publish all the elements from the source and will finally send the
+`onComplete()` signal. This signal will also be offloaded to the `publisherExecutor` and will complete the `Future` with
+the integer collection result.
 
-- Wrap the `Subscriber` such that all `Subscriber` and `Subscription` method calls are offloaded to a chosen
-`Executor`. This mode is for operators that process signals from the original `Publisher` asynchronously.
-- Do not wrap the `Subscriber` and forward all `Subscriber` and `Subscription` methods directly on the calling thread to
-avoid the overhead of wrapping each and every `Subscriber` and `Subscription` in a chain. This mode is for operators
-that process signals from the original `Publisher` synchronously.
+<6> The application thread will wait at `get()` for the `Future` result to be asynchronously completed.
 
-Taking the same example from xref:{page-version}@servicetalk-concurrent-api::blocking-safe-by-default.adoc[here]
+=== `subscribeOn()` Example
+
+Using the `subscribeOn(executor)` operator allows processing of the subscription and demand on a specific thread. Using
+the `subscribeOn(executor)` operator generally requires an understanding of the behavior of the source; using a
```

Comment:
`generally requires an understanding of the behavior of the source` -> I don' think this is accurate. The "should I offload or not" decision is made at subscribe time. If the thread doing the subscribe, or interacting with the subscription is sensitive to blocking (e.g. and EventLoop thread) offloading is used.